Convert replay CLI to typescript

ryanjduffy opened the pull request.

markerikson commented on packages/replay/metadata/index.ts:
```diff
@@ -1,18 +1,28 @@
 const ALLOWED_KEYS = Object.keys(handlers);
 
+function isAllowedKey(key: string): key is AllowedKey {
+  if (ALLOWED_KEYS.includes(key)) {
```
:hand: tiny nitpick: could just be `return ALLOWED_KEYS.includes(key)` :)

markerikson commented on packages/replay/metadata/index.ts:
```diff
@@ -24,9 +34,9 @@ function sanitize(metadata, opts = {}) {
+    } else if (isAllowedKey(key)) {
       // validate known types
-      Object.assign(updated, handlers[key](metadata));
+      Object.assign(updated, handlers[key](metadata as any));
```
:speech_balloon: Probably a better way to do it here, but this is not safety-critical code, so YOLO

ryanjduffy replied:
Agreed. I spent a little time trying to figure out the right type but couldn't and I bailed on it.

markerikson commented on packages/replay/metadata/test.ts:
```diff
@@ -1,3 +1,5 @@
+import { Struct } from "superstruct";
```
:hand: can probably use `import type { Struct }` here. Not critical, but nice to delineate what's a type vs what's a value

markerikson commented on packages/replay/metadata/test.ts:
```diff
@@ -13,30 +15,30 @@ const isUuid = require("is-uuid");
 
-function init(data) {
-  const version = data.version || VERSION;
+function init(data: Record<string, unknown>) {
```
:hand: given that we have `Record<string, unknown>` popping up quite a bit, I'll often extract that into its own type just so I don't repeat myself

markerikson commented on packages/replay/src/upload.ts:
```diff
@@ -1,20 +1,26 @@
         async onOpen() {
           try {
-            await gClient.setAccessToken(accessToken);
+            await gClient?.setAccessToken(accessToken);
```
:hand: We should be pretty sure it exists on this line, right? Looks like `!` would be more appropriate here

markerikson reviewed (approved):
Tiny nitpicks, but LGTM!

ryanjduffy commented:
Fixed up a few things based on @markerikson 's comments in c1b2f60 and then realized there were some ts issues when you bootstrapped lerna so I also added 3cee941 to get everything working so you can:

* `npm run bootstrap` to install all deps and cross-link packages
* `npm run typecheck` to run `tsc` for each bundle`
* and run `npm link` in the `dist/` folder of each package to use the local copy for testing

😓
